test/js-native-api/6_object_wrap/test-basic-finalizer.js

```diff
@@ -0,0 +1,20 @@
+// Flags: --expose-gc
```

Comment:
Should we also test that the finalizer was called? Like, adding a heap-allocated `int` to the environment with `napi_set_instance_data` which is incremented from the finalizer, and an accessor to ensure that the value was incremented. We should try to make sure, if possible, that we check after gc but before the event loop has had a chance to run. That way we can distinguish between synchronous and asynchronous finalization.